gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/QueryAndGatewayDtoConverter.java

```diff
@@ -471,8 +494,20 @@ private static boolean hasTimeRangeFilter(Filter filter, String timestampAttribu
       return filter.getChildFilterList().stream()
           .anyMatch(f -> hasTimeRangeFilter(f, timestampAttributeId));
     }
-    return filter.getLhs().getValueCase() == Expression.ValueCase.COLUMNIDENTIFIER
-        && filter.getLhs().getColumnIdentifier().getColumnName().equals(timestampAttributeId);
+    return getSelectionAttributeId(filter.getLhs())
+        .map(filterAttributeId -> filterAttributeId.equals(timestampAttributeId))
+        .orElse(false);
+  }
+
+  private static Optional<String> getSelectionAttributeId(Expression expression) {
```

Comment:
should we rename this `getSelectionAttributeId` -> `getAttributeId`? The first time, I was reading, I read something to do with the selection clause. What do you think?

Reply:
What I was trying to convey here (and I think is missing from `getAttributeId`)  is that this only extracts attribute id from a col identifier or attribute expression, rather than an order by, function etc. I had taken to lumping the two of them together in naming as "selection", but I can see how that would be unclear. Any ideas that could better convey that distinction?

Reply:
I see it's a bit tricky as the two names are very different 'columnIdentifier' and 'attributeExpression'. Not able to come up with a good name so that it is readable in the above context. Maybe any of the below (if not we can use the same name for now).

`<get|fetch|extract>AttributeIdFromColumnOrAttributeExpression` 
`getAttributeIdFromAttributeSelection` - As we are using, somewhere attributeSelection which refers to Column/AttributeExpression

Reply:
I buy `getAttributeIdFromAttributeSelection` - will update to that.

Reply:
done